src/main/java/ir/ui/golestan/restservice/ProfessorController.java

```diff
@@ -1,24 +1,24 @@
 package ir.ui.golestan.restservice;
 
 
+import org.springframework.http.RequestEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import ir.ui.golestan.data.entity.Score;
 import ir.ui.golestan.data.repository.ScoreRepository;
+import ir.ui.golestan.authorization.AuthenticatedUser;
+import ir.ui.golestan.authorization.Role;
 
 @RestController
 public class ProfessorController {
-    private static final String template = "***User ID: %d, Score: %d, Course ID: %. 2f***";
 
 	@GetMapping("/professor")
-    public void addScore(@RequestParam(value = "studentId") int studentId,
-                         @RequestParam(value = "courseId") int courseId,
-                         @RequestParam(value = "score") double score,
-                         ScoreRepository repository) {        // Repository???
-
-        System.out.println(String.format(template, studentId, courseId, score));
+    public void setStudentScore(RequestEntity<?> request, int studentId, int courseId, 
+                                double score, ScoreRepository repository) {// Repository???
```

Comment:
RestController va har class dige ee ke ba @Component va farzandanesh alamat gozari beshe to spring mishe ye `bean` .
yani spring miad nega be constructor in class mikone va har chi ke lazaem bashe ro behesh mide va ye shei azah dorost mikone va to khodesh negah midare. va baad har jaye dige ye class dige in ro khast behesh mide. alan ma ScoreRepositiry to in class lazem darim. mizarimesh to counstructoresh. zemn in ke controler ha bayad az BaseController ers bebarand ta method getAuthenticatedUser ro betonim estefade konim

Reply:
```suggestion
    public class ProfessorController extends BaseController {

    private final ScoreRepository scoreRepository;

    public ProfessorController(GolestanConfiguration configuration, AuthorizationService authorizationService,
            ScoreRepository scoreRepository) {
        super(configuration, authorizationService);
        this.scoreRepository = scoreRepository;
    }
```
in Okye?

Reply:
are